src/main/java/appeng/client/gui/implementations/GuiPatternMulti.java

```diff
@@ -0,0 +1,252 @@
+    private GuiButton set;
+
+    private GuiButton plus1;
+    private GuiButton plus10;
+    private GuiButton plus100;
+    private GuiButton plus1000;
+    private GuiButton minus1;
+    private GuiButton minus10;
+    private GuiButton minus100;
+    private GuiButton minus1000;
+
+    private GuiBridge originalGui;
+
+    @Reflected
+    public GuiPatternMulti(final InventoryPlayer inventoryPlayer, final ITerminalHost te) {
+        super(new ContainerPatternMulti(inventoryPlayer, te));
+        GuiContainer gui = (GuiContainer) Minecraft.getMinecraft().currentScreen;
+    }
+
+    @Override
+    public void initGui() {
+        super.initGui();
+
+        final int a = AEConfig.instance.craftItemsByStackAmounts(0);
+        final int b = AEConfig.instance.craftItemsByStackAmounts(1);
+        final int c = AEConfig.instance.craftItemsByStackAmounts(2);
+        final int d = AEConfig.instance.craftItemsByStackAmounts(3);
+
+        this.buttonList.add(this.plus1 = new GuiButton(0, this.guiLeft + 20, this.guiTop + 26, 22, 20, "+" + a));
+        this.buttonList.add(this.plus10 = new GuiButton(0, this.guiLeft + 48, this.guiTop + 26, 28, 20, "+" + b));
+        this.buttonList.add(this.plus100 = new GuiButton(0, this.guiLeft + 82, this.guiTop + 26, 32, 20, "+" + c));
+        this.buttonList.add(this.plus1000 = new GuiButton(0, this.guiLeft + 120, this.guiTop + 26, 38, 20, "+" + d));
+
+        this.buttonList.add(this.minus1 = new GuiButton(0, this.guiLeft + 20, this.guiTop + 75, 22, 20, "-" + a));
+        this.buttonList.add(this.minus10 = new GuiButton(0, this.guiLeft + 48, this.guiTop + 75, 28, 20, "-" + b));
+        this.buttonList.add(this.minus100 = new GuiButton(0, this.guiLeft + 82, this.guiTop + 75, 32, 20, "-" + c));
+        this.buttonList.add(this.minus1000 = new GuiButton(0, this.guiLeft + 120, this.guiTop + 75, 38, 20, "-" + d));
+
+        this.buttonList
+                .add(this.set = new GuiButton(0, this.guiLeft + 128, this.guiTop + 51, 38, 20, GuiText.Set.getLocal()));
+        this.buttonList.add(
+                this.symbolSwitch = new GuiImgButton(
+                        this.guiLeft + 22,
+                        this.guiTop + 53,
+                        Settings.ACTIONS,
+                        ActionItems.MULTIPLY));
+
+        ItemStack myIcon = null;
+        final Object target = ((AEBaseContainer) this.inventorySlots).getTarget();
+        final IDefinitions definitions = AEApi.instance().definitions();
+        final IParts parts = definitions.parts();
+
+        if (target instanceof PartPatternTerminal) {
+            for (final ItemStack stack : parts.patternTerminal().maybeStack(1).asSet()) {
+                myIcon = stack;
+            }
+            this.originalGui = GuiBridge.GUI_PATTERN_TERMINAL;
+        }
+
+        if (target instanceof PartPatternTerminalEx) {
+            for (final ItemStack stack : parts.patternTerminalEx().maybeStack(1).asSet()) {
+                myIcon = stack;
```

Comment:
Probably there should be `break` after first valid stack.

Reply:
Yes, but the logic for this code comes from [GuiCraftAmount](https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/blob/3b1c99866b1095d2fad9b777a7876b09e9f781ef/src/main/java/appeng/client/gui/implementations/GuiCraftAmount.java#L109), which doesn't use `break` either, so I don't think `break` is needed